scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroFileOperations.java

```diff
@@ -193,10 +202,15 @@ private static class AvroReader<ValueT> extends FileOperations.Reader<ValueT> {
     public void prepareRead(ReadableByteChannel channel) throws IOException {
       final Schema schema = schemaSupplier.get();
 
-      DatumReader<ValueT> datumReader =
-          recordClass == null
-              ? new GenericDatumReader<>(schema)
-              : new ReflectDatumReader<>(recordClass);
```

Comment:
SMB avro read was not symmetric with writer and did not have any logical type conversion applied.

Reply:
We were lucky nobody used logical types in SMB?

Reply:
Looks like it yes